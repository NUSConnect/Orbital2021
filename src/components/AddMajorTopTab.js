import React from 'react'
import { StyleSheet, Text, View, TouchableOpacity } from 'react-native'
import { Ionicons } from 'react-native-vector-icons'

export default function AddMajorTopTab ({ onPress }) {
  return (
    <View style={styles.header}>
      <TouchableOpacity
        style={styles.icon}
        activeOpacity={0.4}
        onPress={onPress}
        testID='back'
      >
        <Ionicons
          name='arrow-back'
          color='#79D2E6'
          size={38}
        />
      </TouchableOpacity>
      <Text style={styles.title}> Add your major </Text>
    </View>
  )
}

const styles = StyleSheet.create({
  title: {
    height: 60,
    lineHeight: 60,
    width: '100%',
    backgroundColor: '#ffffff',
    color: '#ff8c00',
    fontSize: 30,
    paddingLeft: 15,
    marginBottom: 10
  },
  header: {
    flexDirection: 'row'
  },
  icon: {
    marginTop: 10
  }
})
