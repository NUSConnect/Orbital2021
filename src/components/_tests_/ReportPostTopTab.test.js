import * as React from 'react'
import AddMajorTopTab from '../AddMajorTopTab'
import renderer from 'react-test-renderer'
import { fireEvent, render } from '@testing-library/react-native'

jest.useFakeTimers()
jest.mock('react-native-vector-icons')

it('renders correctly', () => {
  const tree = renderer.create(<AddMajorTopTab />)
  expect(tree).toMatchSnapshot()
})

it('on press button', () => {
  const mockOnPressFn = jest.fn()
  const { getByTestId } = render(<AddMajorTopTab onPress={mockOnPressFn} />)
  fireEvent.press(getByTestId('back'))
  expect(mockOnPressFn).toHaveBeenCalled()
})
