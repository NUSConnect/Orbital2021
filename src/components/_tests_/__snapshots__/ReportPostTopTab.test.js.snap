// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginTop": 10,
        "opacity": 1,
      }
    }
    testID="back"
  />
  <Text
    style={
      Object {
        "backgroundColor": "#ffffff",
        "color": "#ff8c00",
        "fontSize": 30,
        "height": 60,
        "lineHeight": 60,
        "marginBottom": 10,
        "paddingLeft": 15,
        "width": "100%",
      }
    }
  >
     Add your major 
  </Text>
</View>
`;
